stdlib/2/collections.pyi

```diff
@@ -93,10 +93,13 @@ class Counter(Dict[_T, int], Generic[_T]):
     def __iand__(self, other: Counter[_T]) -> Counter[_T]: ...
     def __ior__(self, other: Counter[_T]) -> Counter[_T]: ...
 
+_OrderedDictT = TypeVar('_OrderedDictT', bound=OrderedDict)
+
 class OrderedDict(Dict[_KT, _VT], Reversible[_KT], Generic[_KT, _VT]):
     def popitem(self, last: bool = ...) -> Tuple[_KT, _VT]: ...
     def __reversed__(self) -> Iterator[_KT]: ...
-    def __copy__(self) -> OrderedDict[_KT, _VT]: ...
+    def __copy__(self: _OrderedDictT) -> _OrderedDictT: ...
+    def copy(self: _OrderedDictT) -> _OrderedDictT: ...
```

Comment:
Note that there are some bugs with self-types in generic types (in mypy). But hopefully most of them will be fixed soon (as soon as I will have time to finish a patch I am working on).